### PR TITLE
Have ssl mode OPTIONAL to optionally allow for ssl

### PR DESCRIPTION
**Pull request by adizaimi**
Current mode ALLOW will try to load ssl cert file at start and exit with failure if can't find that file.
This mode, OPTIONAL, tries to load cert file, and if it does not find it it continues starting up the db.
This mode is useful for tests when run with SKIPSSL=1 option. 
We will not document this because it does not seem useful in any other circumstance to use this mode.

### CODE REVIEW

**Comment by riverszhang89** [comment]
Could you please update the docs too.

**Comment by adizaimi** [comment]
I'm not sure whether we want to advertise that option, but sure can do.

File: docs/pages/config/ssl.md [hunk 1]
```diff
@@ -152,13 +152,14 @@ are refused to join the cluster.
...
-| `ssl_client_mode mode` | Can be one of `ALLOW`, `REQUIRE`, `VERIFY_CA` and `VERIFY_HOSTNAME` | `ALLOW` |
+| `ssl_client_mode mode` | Can be one of `ALLOW`<sup>[1](#ssloptfootnote)</sup>, `REQUIRE`, `VERIFY_CA` and `VERIFY_HOSTNAME` | `ALLOW` |
 | `ssl_replicant_mode mode` | Can be one of `ALLOW`, `REQUIRE`, `VERIFY_CA` and `VERIFY_HOSTNAME` | `ALLOW` |
 | `ssl_cert_path path` | Directory containing the server certificate files. Comdb2 searches `server.crt`, `server.key` and `root.crt` for the server certificate, key and trusted CAs respectively | The data directory |
 | `ssl_cert file`| Path to the certificate | `<ssl_cert_path>/server.crt` |
 | `ssl_key file` | Path to the key | `<ssl_cert_path>/server.key` |
 | `ssl_ca file` | Path to the trusted CA certificates | `<ssl_cert_path>/root.crt` |
 
+<a name="ssloptfootnote">[1]</a>: If mode is `ALLOW`, the server will try to open the certificate file in the designated location and will fail to start if it can not be found. For te server configuration, an equivalent mode to `ALLOW` is `OPTIONAL` which behaves in exactly the same way exept it will not fail to start the server if certificate file is not found at start time.
```
Comment (nirbhayc) [inline docs/pages/config/ssl.md]: Spell checks:
For te server -> For the server
exept -> except

Comment (mponomar) [inline docs/pages/config/ssl.md]: make spellcheck in docs/, thanks to Patrick

File: docs/pages/config/ssl.md [hunk 2]
```diff
@@ -152,13 +152,14 @@ are refused to join the cluster.
...
-| `ssl_client_mode mode` | Can be one of `ALLOW`, `REQUIRE`, `VERIFY_CA` and `VERIFY_HOSTNAME` | `ALLOW` |
+| `ssl_client_mode mode` | Can be one of `ALLOW`<sup>[1](#ssloptfootnote)</sup>, `REQUIRE`, `VERIFY_CA` and `VERIFY_HOSTNAME` | `ALLOW` |
 | `ssl_replicant_mode mode` | Can be one of `ALLOW`, `REQUIRE`, `VERIFY_CA` and `VERIFY_HOSTNAME` | `ALLOW` |
 | `ssl_cert_path path` | Directory containing the server certificate files. Comdb2 searches `server.crt`, `server.key` and `root.crt` for the server certificate, key and trusted CAs respectively | The data directory |
 | `ssl_cert file`| Path to the certificate | `<ssl_cert_path>/server.crt` |
 | `ssl_key file` | Path to the key | `<ssl_cert_path>/server.key` |
 | `ssl_ca file` | Path to the trusted CA certificates | `<ssl_cert_path>/root.crt` |
 
+<a name="ssloptfootnote">[1]</a>: If mode is `ALLOW`, the server will try to open the certificate file in the designated location and will fail to start if it can not be found. For the server configuration, an equivalent mode to `ALLOW` is `OPTIONAL` which behaves in exactly the same way except it will not fail to start the server if certificate file is not found at start time.
```
Comment (riverszhang89) [inline docs/pages/config/ssl.md]: This isn't quite accurate. `OPTIONAL` mode can still fail to proceed if certificate locations are explicitly specified and certificate files could not be found there. In other words, specifying any of `ssl_cert_path`, `ssl_cert`, `ssl_key` and `ssl_ca` implies `ALLOW`.

**Comment by riverszhang89** [comment]
Per discussion, let's undocument `OPTIONAL`.

**Pull request merged**